python/nav/web/api/v1/views.py

```diff
@@ -764,6 +765,28 @@ class VlanViewSet(NAVAPIMixin, viewsets.ModelViewSet):
     search_fields = ['net_ident', 'description']
 
 
+class PrefixFilterClass(FilterSet):
+    contains_address = CharFilter(method="contains_address_filter")
+
+    def contains_address_filter(self, queryset, field_name, value):
+        if not value:
+            return queryset
+        if not is_valid_cidr(value):
+            raise ValidationError(
+                {"contains_address": ["Value must be a valid CIDR address"]}
+            )
```

Comment:
For the other filters, vlan for instance, the error message returned from the API looks like {"vlan": ["error message"]},
so to replicate that and keep it consistent I did it like this. If you just do 
```
raise ValidationError("Value must be a valid CIDR address")
```
Then the return you get from the API will be  ["Value must be a valid CIDR address"], not containing the filter name itself.

This solution is not optimal, I would like for the filter name `contains_address` to not be hard coded in the error message, but not sure how to cleanly do that.

Reply:
Error reporting from the API is, unfortunately, not done in a very consistent manner. I think we need a bigger cleanup process for that, so keeping some sense of compatibility with what was already there is probably fine in this case.

Reply:
Yeah, argus has the same problem. DRF isn't consistent with this, or strictly speaking: a bit too fancy!